source/daq/frequency_transform.cc

```diff
@@ -0,0 +1,271 @@
+                out_stream< 0 >().data()->set_pkt_in_session( 0 );
+                out_stream< 1 >().data()->set_pkt_in_session( 0 );
+                if( ! out_stream< 0 >().set( stream::s_start ) ) return;
+                if( ! out_stream< 1 >().set( stream::s_start ) ) return;
+            }
+
+            try
+            {
+                LPROG( plog, "Starting main loop (frequency transform)" );
+                while (! is_canceled() )
+                {
+                    if (out_stream< 0 >().get() == stream::s_stop || out_stream< 1 >().get() == stream::s_stop)
+                    {
+                        LWARN( plog, "output stream(s) have stop condition" );
+                        break;
+                    }
+                    // check for [un]pause instruction
+                    if( have_instruction() )
+                    {
+                        if( f_paused && use_instruction() == midge::instruction::resume )
+                        {
+                            LDEBUG( plog, "freq transform resuming" );
+                            if( ! out_stream< 0 >().set( stream::s_start ) ) throw midge::node_nonfatal_error() << "Stream 0 error while starting";
+                            if( ! out_stream< 1 >().set( stream::s_start ) ) throw midge::node_nonfatal_error() << "Stream 1 error while starting";
+                            f_paused = false;
+                        }
+                        else if( !f_paused && use_instruction() == midge::instruction::pause )
+                        {
+                            LDEBUG( plog, "freq transform pausing" );
+                            if( ! out_stream< 0 >().set( stream::s_stop ) ) throw midge::node_nonfatal_error() << "Stream 0 error while stopping";
+                            if( ! out_stream< 1 >().set( stream::s_stop ) ) throw midge::node_nonfatal_error() << "Stream 1 error while stopping";
+                            f_paused = true;
+                        }
+                    }
+
+                    if ( !f_paused )
+                    {
+                        time_data_in = in_stream< 0 >().data();
+                        time_data_out = out_stream< 0 >().data();
+                        freq_data_out = out_stream< 1 >().data();
+                        LDEBUG( plog, "next steams acquired, doing FFT" );
+
+                        //time output
+                        *time_data_out = *time_data_in;
+                        //frequency output
+                        std::copy(&time_data_in->get_array()[0][0], &time_data_in->get_array()[0][0] + f_fft_size*2, &f_fftw_input[0][0]);
+                        fftw_execute_dft(f_fftw_plan, f_fftw_input, f_fftw_output);
+                        //is this the normalization we want? (is it what the ROACH does?)
+                        for (size_t i_bin=0; i_bin<f_fft_size; ++i_bin)
+                        {
+                            f_fftw_output[i_bin][0] *= sqrt(1. / (double)f_fft_size);
```

Comment:
Define a local variable at the top level of this function (or the widest scope in which `f_fft_size` doesn't change): `double scaling = sqrt(1. / (double)f_fft_size);`.  Both `sqrt` and division are relatively slow operations, so you definitely want to avoid doing it for each bin, and even for each record.